### ISSUE TITLE
OPAM for Coq dev

### ISSUE THREAD

**Issue by liyishuai** [PrincetonUniversity/VST#453]
Projects like [_Software Foundations_](https://github.com/DeepSpec/sf) depend on VST and hope to be tested against Coq dev. The OPAM package for VST depends on `coq-platform` variants of CompCert, which currently supports Coq <= 8.12. Should we create a `coq-platform` variant of CompCert on `coq-extra-dev` that supports Coq dev? @MSoegtropIMC

**Comment by MSoegtropIMC** [comment]
Yes, it would make sense to lift the 8.12 restriction. If it makes sense to test against CompCert master rather than the latest release is a different question. Since CompCert is quite stable, probably yes, but the platform CompCert patches (there are quite a few) might break easily.

**Comment by liyishuai** [comment]
As a user I'm fine with any VST and CompCert that works with Coq dev. If the CompCert master branch breaks too quickly, we can release `coq-compcert.DATE~coq-platform` pinned to a specific CompCert commit, and only create a new version when a change in Coq dev requires an updated CompCert.

**Comment by MSoegtropIMC** [comment]
Yes, this makes sense. I would use a non release version of CompCert only if either Coq or VST master compatibility require it.

**Comment by andrew-appel** [comment]
@liyishuai , I have increased your VST repo privileges from "Read" to "Write", which can make it slightly more convenient for you to submit pull requests without having to fork the repo.  If you would like to propose a pull-request that adjusts the Makefile or other parts of the build as you suggest, you may go ahead.

**Comment by andrew-appel** [comment]
After discussion with @MSoegtropIMC it seems that we should have a "dev" branch of VST that tracks the master branch of CompCert.  The main purpose of this is to include in the continuous integration for the "Coq Platform", which also tracks the master branch of CompCert.  Most pull requests (which are likely quite independent of the usually minor differences between CompCert latest-release and CompCert master) can be merged to master and pulled to "dev".  Or I supposed they could be merged to dev and cherry-picked to master, but that seems less straightforward.  Before I implement this, does anyone have any comment?

**Comment by MSoegtropIMC** [comment]
I would not recommend to have a separate master and dev branch where all commits are merged on - this might be a lot of work. I would rather have a master branch which is kept up to date with CompCert master and VST release branches which are for specific VST releases (like 2.7.X) and are tied to specific CompCert releases.

Long term developments like VSU should happen on a separate development branch (called e.g. `vsu`) and be merged over after important milestones to master and possibly a release branch. Doing such developments on master is inconvenient, because one wants to do such developments in a stable environment (with a release compcert).

Having said that we are almost back to having a master branch using a release CompCert, where the majority of development happens, and a dev branch which is tied to compcert master - only the names are swapped. Typically a development branch is where the development happens and the master branch is where the big feature milestones and e.g. CompCert compatibility changes are merged together.

The opam .dev packages are named a bit unfortunate - they all point to master branches and not to development branches.

**Comment by MSoegtropIMC** [comment]
@liyishuai : Since #485 is now fixed, I will publish a VST.dev opam package soon (cause of #485 such a package did not work before).

Since currently VST master compiles with CompCert master, we can postpone the discussion on what to do if this stops working when it stops working, since this is easier to discuss on a concrete case.

**Comment by andrew-appel** [comment]
Question for @liyishuai :  Is this situation now satisfactorily resolved, to the point where dev branches of Software Foundations can be tested against CompCert and VST?

Remark:  As a matter of general policy, VST master is not always guaranteed to compile against CompCert master.  The VST master is supposed to compile against the released version of CompCert.  When there are incompatible changes to CompCert that are not yet CompCert "releases", I will often create a VST "Draft Pull Request" that is compatible with the CompCert master.  Two examples of those are  #522 and #578 .
